### ISSUE TITLE
Services / Unbound DNS - unbound-anchor for exclusive DNS over TLS setups

### ISSUE THREAD

**Issue by AdSchellevis** [opnsense/core#5441]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

After switching Unbound to use TLS only and placing a machine in an environment which doesn't allow UDP port 53 traffic, restarts of unbound take ages. Apparently `unbound-anchor` wil try to fetch the trusted root certificates over UDP/53, which isn't possible due to policy constraints. 

https://github.com/opnsense/core/blob/72e6ca2feed1f2714e63cd03d9ee92e6ce212601/src/opnsense/scripts/unbound/start.sh#L33

**To Reproduce**

Configure OPNsense to use DNS over TLS and enable DNSSEC, make sure port UDP/53 traffic isn't possible and restart Unbound, since the `/var/unbound/root.key` file is likely still there, eventually it will start, but it will take a long time. Removing `/var/unbound/root.key` should probably result in a non functional unbound (haven't tried, but sounds logical). 

**Expected behavior**

I'm not entirely sure how this is supposed to work, but Unbound shouldn't rely on UDP/53 in these cases. 

**Describe alternatives you considered**

none

**Additional context**

.

**Environment**

Software version used and hardware type if relevant, e.g.:

OPNsense 21.7.7 (amd64, OpenSSL).

**Comment by swhite2** [comment]
There are actually multiple sides to this problem. According to the documentation provided in https://github.com/NLnetLabs/unbound/blob/master/smallapp/unbound-anchor.c, `unbound-anchor` will use DNSKEY UDP queries to verify that the current local root certificate is valid - and update the root anchor file accordingly. If this fails, an RFC7958 built-in certificate is fetched over HTTPS and replaces the current local certificate.

This immediately poses a problem, since both the DNSKEY UDP queries and fetching the nameserver for a HTTPS upgrade requires unencrypted DNS over port 53, meaning both the main update policy and failover policy fail. In this specific case, the program does nothing and assumes the certificate is still valid - which it is, until it isn't when it expires, or so I assume. This is reflected in the documentation: 
```
By default it prints almostnothing.	
It also prints nothing on errors by default;  in that
case the original root anchor file is simply left undisturbed,
so that a recursive server can start right after it.
```

Furthermore, if you have a first-line defense setup in which unbound-anchor is not run on the same machine as the machine on which port 53 UDP is blocked, `unbound-anchor` will block until it decides it hasn't received an answer, which may run for a very long time - hence the boot issues. A quick fix (and I confirmed this to be the case), would be to also block UDP port 53 locally, in which case the sendmsg() syscall will immediately return "Permission denied". This at least solves the long boottime issue.

It seems to me though, that when explicitly using port 853 for DNS over TLS, unbound-anchor should also be using this to resolve it's nameservers. I did a quick test to see if adding the failover nameserver (data.iana.org) as a raw IP address to the system hosts file would resolve the failover scenario, which it did at first but I quickly realised I had temporarily disabled the UDP block rule. It seems unbound-anchor still requires resolving multiple hostnames, even in a failover scenario.
